aspnetcore/tutorials/build-your-first-razor-components-app.md

```diff
@@ -0,0 +1,188 @@
+
+1. Provide the initial markup for the page:
+
+   ```cshtml
+   @page "/todo"
+
+   <h1>Todo</h1>
+   ```
+
+1. Add the Todo page to the navigation bar by updating *WebApplication1.App/Shared/NavMenu.cshtml*. Add a `<NavLink>` for the Todo page by adding the following list item markup below the existing list items.
+
+   ```cshtml
+   <li class="nav-item px-3">
+       <NavLink class="nav-link" href="todo">
+           <span class="oi oi-list-rich" aria-hidden="true"></span> Todo
+       </NavLink>
+   </li>
+   ```
+
+1. Refresh the app in the browser. Visit the new Todo page to confirm that the link to the Todo page works.
+
+1. Add a *TodoItem.cs* file to the root of the project to hold a class that represents a todo item. Use the following C# code for the `TodoItem` class:
+
+   [!code-cshtml[](build-your-first-razor-components-app/samples/3.x/WebApplication1/WebApplication1.App/TodoItem.cs)]
+
+1. Return to the Todo component (*Todo.cshtml*):
+
+   * Add a field for the todos in an `@functions` block. The Todo component uses this field to maintain the state of the todo list.
+   * Add unordered list markup and a `foreach` loop to render each todo item as a list item.
+
+   [!code-cshtml[](build-your-first-razor-components-app/samples_snapshot/3.x/FetchData4.cshtml?highlight=5-10,12-14)]
+
+1. The app requires UI elements for adding todos to the list. Add a text input and a button below the list:
+
+   [!code-cshtml[](build-your-first-razor-components-app/samples_snapshot/3.x/FetchData5.cshtml?highlight=12-13)]
+
+1. Refresh the browser. When the **Add todo** button is selected, nothing happens because an event handler isn't wired up to the button.
+
+1. Add an `AddTodo` method to the `Todo` component and register it for button clicks using the `onclick` attribute:
+
+   [!code-cshtml[](build-your-first-razor-components-app/samples_snapshot/3.x/FetchData6.cshtml?highlight=2,7-10)]
+
+   The `AddTodo` C# method is called when the button is selected.
+
+1. To get the title of the new todo item, add a `newTodo` string field and bind it to the value of the text input using the `bind` attribute:
+
+   [!code-cshtml[](build-your-first-razor-components-app/samples_snapshot/3.x/FetchData7.cshtml?highlight=2)]
+
+   ```cshtml
+   <input placeholder="Something todo" bind="@newTodo" />
+   ```
+
+1. Update the `AddTodo` method to add the `TodoItem` with the specified title to the list. Clear the value of the text input by setting `newTodo` to an empty string:
+
+   [!code-cshtml[](build-your-first-razor-components-app/samples_snapshot/3.x/FetchData8.cshtml?highlight=19-26)]
+
+1. Refresh the browser. Add some todos to the todo list to test the new code.
+
+1. The title text for each todo item can be made editable as well. Add a check box input for each todo item and bind its value to the `IsDone` property. Change `@todo.Title` to an `<input>` element bound to `@todo.Title`:
```

Comment:
> The title text for each todo item can be made editable as well.

as well as what?

Reply:
Yep ... this got *bass-ackwards* :smile: with edits over time. It was originally from the blog post ... paragraph 9 of https://blogs.msdn.microsoft.com/webdev/2018/03/22/get-started-building-net-web-apps-in-the-browser-with-blazor/#build-a-todo-list

It was originally referring to something editable in addition to the check boxes.

I'll remove "as well."